src/main/kotlin/net/atos/zac/smartdocuments/rest/RESTSmartDocumentsTemplateGroup.kt

```diff
@@ -16,3 +16,20 @@ data class RESTSmartDocumentsTemplateGroup(
     var groups: Set<RESTSmartDocumentsTemplateGroup>?,
     var templates: Set<RESTSmartDocumentsTemplate>?,
 )
+
+fun Set<RESTSmartDocumentsTemplateGroup>.toStringRepresentation(): Set<String> {
+    val result = mutableSetOf<String>()
+    this.forEach { result.addAll(convertTemplateGroupToStringRepresentation(it, null)) }
+    return result
+}
+
+private fun convertTemplateGroupToStringRepresentation(
+    group: RESTSmartDocumentsTemplateGroup,
+    parent: String?
+): Set<String> =
+    arrayOf(parent, "group.${group.id}.${group.name}").filterNotNull().joinToString(".").let { groupString ->
+        mutableSetOf(groupString).apply {
```

Comment:
also no need for a mutable set I think

Reply:
We need it here as the lambda uses `mapTo(this)`

Reply:
ok, clear! I think it should still be possible to do this all without a mutable set but maybe for another day. :-)

Reply:
The main problem is that this is a recursive function, that collects data across different calls.

Reply:
ah yes of course.. clear.